exasol/toolbox/nox/tasks.py

```diff
@@ -323,3 +332,127 @@ def report(session: Session) -> None:
+
+        version: A version string of the following format: {number}.{number}.{number} (Major, Minor, Patch).
+    """
+
+    def _parser():
+        parser = argparse.ArgumentParser(
+            prog=f"nox -s prepare-release",
+            usage="nox -s prepare-release -- [-h] version",
+            formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        )
+        parser.add_argument(
+            "version",
+            type=version,
+            help=("A version string of the following format:" '"NUMBER.NUMBER.NUMBER"'),
+        )
+        parser.add_argument(
+            "--no-add",
+            default=False,
+            action="store_true",
+            help=("Neither add nor commit the changes"),
+        )
+        parser.add_argument(
+            "--no-branch",
+            default=False,
+            action="store_true",
+            help=("Do not create a branch to commit the changes on"),
+        )
+        parser.add_argument(
+            "--no-pr",
+            default=False,
+            action="store_true",
+            help=("Do not create a pull request for the changes"),
+        )
+        return parser
+
+    parser = _parser()
+    args = parser.parse_args(session.posargs)
+    new_version = args.version
+    old_version = Version.from_poetry()
+    if not new_version >= old_version:
+        error_msg = (
+            f"Invalid version: the release version ({new_version}) "
+            f"must be greater than or equal to the current version ({old_version})."
+        )
+        session.error(error_msg.format(new=new_version, old=old_version))
+
+    if not args.no_branch and not args.no_add:
+        # prepare branch
+        session.run("git", "switch", "-c", f"release/prepare-{new_version}")
+
+    # bump project version and sync version file
+    session.run("poetry", "version", f"{new_version}")
+    _version(session, Mode.Fix, PROJECT_CONFIG.version_file)
+
+    # create a changelog file for the release and also create a new empty unrleased file
+    unreleased = Path(PROJECT_CONFIG.root) / "doc" / "changes" / "unreleased.md"
+    changelog = (
+        Path(PROJECT_CONFIG.root) / "doc" / "changes" / f"changes_{new_version}.md"
+    )
+    changes = Path(PROJECT_CONFIG.root) / "doc" / "changes" / f"changelog.md"
+
+    changelog_content = extract_release_notes(unreleased)
+    changelog.write_text(new_changelog(new_version, changelog_content))
+
+    unreleased.write_text(new_unreleased())
+
+    changes_content = new_changes(changes, new_version)
+    changes.write_text(changes_content)
+
+    if args.no_add:
+        return
+
+    # 3. commit changes
+    session.run("git", "add", f"{changelog}")
+    session.run("git", "add", f"{unreleased}")
+    session.run("git", "add", f"{changes}")
+    session.run("git", "add", f"{PROJECT_CONFIG.root / 'pyproject.toml'}")
+    session.run("git", "add", f"{PROJECT_CONFIG.version_file}")
+    session.run("git", "commit", "-m", f"Prepare release {new_version}")
```

Comment:
this look like it should go into its own function

Reply:
Yes, I would still like to do that as a follow-up, due to the fact that I planed to clean up the modules in a PR after merging this PR. I think we've reached a point where we can do so in tasks.py, as well as in a few other modules.